src/test/java/org/json/junit/JSONObjectTest.java

```diff
@@ -72,6 +67,253 @@ public class JSONObjectTest {
+        assertEquals("{}", transformedMap.get("key8"));
+    }
+
+    /**
+     * Test Case 2.2.2: Use the stream() method to find the entry with the
+     * minimum and maximum value based on a comparator
+     */
+    @Test
+    public void test_2_2_2() {
+        String data = "{\n" +
+                "  \"key1\": 3,\n" +
+                "  \"key2\": 42,\n" +
+                "  \"key3\": 15,\n" +
+                "  \"key4\": 8,\n" +
+                "}\n";
+
+        JSONObject jsonObject = new JSONObject(data);
+        Comparator<Map.Entry<String, Object>> valueComparator = Comparator.comparingInt(entry -> (Integer) entry.getValue());
+
+        Optional<Map.Entry<String, Object>> minEntry = jsonObject.stream()
+                .min(valueComparator);
+        Optional<Map.Entry<String, Object>> maxEntry = jsonObject.stream()
+                .max(valueComparator);
+
+        System.out.println("Min entry: " + minEntry);
+        System.out.println("Max entry: " + maxEntry);
+        assertTrue(minEntry.isPresent());
+        assertEquals("key1: 3", minEntry.get().getKey() + ": " + minEntry.get().getValue());
+        assertTrue(maxEntry.isPresent());
+        assertEquals("key2: 42", maxEntry.get().getKey() + ": " + maxEntry.get().getValue());
+    }
+
+    /**
+     * Test Case 2.2.3: Use the stream() method to collect entries into a map grouping them
+     * by a classification function based on the keys
+     */
+    @Test
+    public void test_2_2_3() {
+        String data = "{\n" +
+                "  \"akey1\": \"abc\",\n" +
+                "  \"bkey1\": 42,\n" +
+                "  \"akeyA\": true,\n" +
+                "  \"bkeyA\": null,\n" +
+                "  \"akeyX\": [1,2,3],\n" +
+                "  \"bkeyX\": {\"a\":\"b\"}\n" +
+                "}\n";
+
+        JSONObject jsonObject = new JSONObject(data);
+        Map<Character, List<Map.Entry<String, Object>>> groupedByInitial = jsonObject.stream()
+                .collect(Collectors.groupingBy(entry -> entry.getKey().charAt(4)));
+
+        System.out.println(groupedByInitial);
+        assertEquals(3, groupedByInitial.size());
+        assertEquals(2, groupedByInitial.get('1').size());
+        assertEquals(2, groupedByInitial.get('A').size());
+        assertEquals(2, groupedByInitial.get('X').size());
+    }
+
+    /**
+     * Test Case 2.3.1: Test the behavior of the stream() method when applied to an empty JSONObject
+     */
+    @Test
+    public void test_2_3_1() {
+        JSONObject jsonObject = new JSONObject();
+        List<Map.Entry<String, Object>> entriesList = jsonObject.stream()
+                .collect(Collectors.toList());
+
+        System.out.println(entriesList);
+        assertTrue(entriesList.isEmpty());
+    }
+
+    /**
+     * Test Case 2.3.2: stream a null JSONObject
+     */
+    @Test
+    public void test_2_3_2() {
+        JSONObject jsonObject = null;
+
+        assertThrows(NullPointerException.class, () -> {
```

Comment:
I don't think this test is adding anything here. It seems obvious that an NPE would be thrown in this case.

Reply:
Removed